haas/ext/auth/database.py

```diff
@@ -131,6 +131,21 @@ def user_remove_project(user, project):
     db.session.commit()
 
 
+@rest_call('PUT', '/auth/basic/user/<user>/set_privilege', Schema({
+    'user': basestring,
+    'privilege': bool,
+}))
+def user_set_privilege(user, privilege):
+    get_auth_backend().require_admin()
+    user = api._must_find(User, user)
+    if privilege == user.is_admin:
+        status = 'admin' if privilege else 'regular'
+        raise DuplicateError('No change. User %s privilege status: %s' %
+                             (user.label, status))
```

Comment:
I would just get rid of this if statement entirely; the api call gets much more readable without it (though the readability could probably be improved otherwise).